src/main/java/de/hysky/skyblocker/skyblock/chocolatefactory/ChocolateFactorySolver.java

```diff
@@ -0,0 +1,165 @@
+			if (i != lore.size() - 1) stringBuilder.append(" ");
+		}
+		return stringBuilder.toString();
+	}
+
+	private Optional<Rabbit> getCoach(ItemStack cpsItem, ItemStack coachItem) {
+		if (coachItem.isEmpty() || cpsItem.isEmpty() || coachItem.getItem() != Items.PLAYER_HEAD || cpsItem.getItem() != Items.COCOA_BEANS) return Optional.empty();
+		String cpsItemLore = getConcattedLore(cpsItem);
+
+		Matcher cpsMatcher = CPS_PATTERN.matcher(cpsItemLore);
+		OptionalDouble currentCps = getDoubleFromMatcher(cpsMatcher);
+		if (currentCps.isEmpty()) return Optional.empty();
+
+		Matcher multiplierMatcher = TOTAL_MULTIPLIER_PATTERN.matcher(cpsItemLore);
+		OptionalDouble totalMultiplier = getDoubleFromMatcher(multiplierMatcher, cpsMatcher.end());
+		if (totalMultiplier.isEmpty()) return Optional.empty();
+
+		String coachLore = getConcattedLore(coachItem);
+
+		Matcher multiplierIncreaseMatcher = MULTIPLIER_INCREASE_PATTERN.matcher(coachLore);
+		OptionalDouble currentCpsMultiplier = getDoubleFromMatcher(multiplierIncreaseMatcher);
+		if (currentCpsMultiplier.isEmpty()) return Optional.empty();
+
+		OptionalDouble nextCpsMultiplier = getDoubleFromMatcher(multiplierIncreaseMatcher);
+		if (nextCpsMultiplier.isEmpty()) { //This means that the coach isn't hired yet.
+			nextCpsMultiplier = currentCpsMultiplier; //So the first instance of the multiplier is actually the amount we'll get upon upgrading.
+			currentCpsMultiplier = OptionalDouble.of(0.0); //And so, we can re-assign values to the variables to make the calculation more readable.
+		}
+
+		Matcher costMatcher = COST_PATTERN.matcher(coachLore);
+		OptionalInt cost = getIntFromMatcher(costMatcher, multiplierIncreaseMatcher.end()); //Cost comes after the multiplier line
+		if (cost.isEmpty()) return Optional.empty();
+
+		return Optional.of(new Rabbit(currentCps.getAsDouble() / totalMultiplier.getAsDouble() * (nextCpsMultiplier.getAsDouble() - currentCpsMultiplier.getAsDouble()), cost.getAsInt()));
+	}
+
+	private Optional<Rabbit> getRabbit(ItemStack item) {
+		String lore = getConcattedLore(item);
+		Matcher cpsMatcher = CPS_INCREASE_PATTERN.matcher(lore);
+		OptionalInt currentCps = getIntFromMatcher(cpsMatcher);
+		if (currentCps.isEmpty()) return Optional.empty();
+		OptionalInt nextCps = getIntFromMatcher(cpsMatcher);
+		if (nextCps.isEmpty()) {
+			nextCps = currentCps; //This means that the rabbit isn't hired yet.
+			currentCps = OptionalInt.of(0); //So the first instance of the cps is actually the amount we'll get upon hiring.
+		}
+
+		Matcher costMatcher = COST_PATTERN.matcher(lore);
+		OptionalInt cost = getIntFromMatcher(costMatcher, cpsMatcher.end()); //Cost comes after the cps line
+		if (cost.isEmpty()) return Optional.empty();
+		return Optional.of(new Rabbit(nextCps.getAsInt() - currentCps.getAsInt(), cost.getAsInt()));
+	}
+
+	private OptionalLong getTotalChocolate(ItemStack item) {
+		if (item.isEmpty() || item.getItem() != Items.PLAYER_HEAD) return OptionalLong.empty();
```

Comment:
```suggestion
		if (!item.isOf(Items.PLAYER_HEAD)) return OptionalLong.empty();
```
same as before, use of isOf and removal of isEmpty